### PR TITLE
NRPT-413: Adds Link Indicator

### PR DESCRIPTION
**Pull request by KitArmstrong**
Ticket: https://bcmines.atlassian.net/browse/NRPT-413

Adds a link indicator icon to the collections detail and edit screens.

### CODE REVIEW

File: api/src/controllers/search.js [hunk 1]
```diff
@@ -747,6 +747,73 @@ const executeQuery = async function (args, res, next) {
...
+          addedBy: 1,
+          dateUpdated: 1,
+          updatedBy: 1,
+          datePublished: 1,
+          publishedBy: 1,
+          isBcmiPublished: 1,
+          collectionRecords: {
+            $filter: { input: "$collectionRecords", as: "a", cond: { $ifNull: ["$$a._id", false] } }
+          }
+        }
```
Comment (danieltruong) [inline api/src/controllers/search.js:816]: Wonder if there's any way to get rid of the group and project?

**Review by danieltruong — COMMENTED** [review]
image
When staging a new records with a link, the list does not show the link icon for the record.

File: angular/projects/admin-nrpti/src/app/mines/mines-collection-detail/mines-collection-detail-record-row/mines-collection-detail-record-row.component.html [hunk 2]
```diff
@@ -1,4 +1,6 @@
-<td scope="row" data-label="Name" class="col-6">{{ rowData.recordName || '-' }}</td>
+<td scope="row" data-label="Name" class="col-6 d-flex">{{ rowData.recordName || '-' }}
+  <i *ngIf="rowData.isLink" class="material-icons ml-2">link</i>
```
Comment (danieltruong) [inline angular/projects/admin-nrpti/src/app/mines/mines-collection-detail/mines-collection-detail-record-row/mines-collection-detail-record-row.component.html:2]: Maybe we should have a document icon if the row is not a link.

**Review by danieltruong — COMMENTED** [review]
image
When staging a new records with a link, the list does not show the link icon for the record.

File: angular/projects/admin-nrpti/src/app/mines/mines-collection-detail/mines-collection-detail-record-row/mines-collection-detail-record-row.component.html (hunk 2 above)
Comment (danieltruong) [inline angular/projects/admin-nrpti/src/app/mines/mines-collection-detail/mines-collection-detail-record-row/mines-collection-detail-record-row.component.html:2]: For some reason sonarcloud is obsessed with changing i tags to em tags...

File: angular/projects/admin-nrpti/src/app/mines/mines-collections-add-edit/mines-collections-add-edit.component.html [hunk 3]
```diff
@@ -152,8 +152,9 @@ <h3 class="border-0 mb-0">Associated Records</h3>
                   <i class="material-icons" cdkDragHandle>reorder</i>
                 </button>
               </td>
-              <td data-label="Record Name" class="col-3">
+              <td data-label="Record Name" class="col-3 d-flex">
                 {{ formGroup.get('record').value.recordName || '-' }}
+                <i *ngIf="formGroup.get('record').value.isLink" class="material-icons ml-2">link</i>
```
Comment (danieltruong) [inline angular/projects/admin-nrpti/src/app/mines/mines-collections-add-edit/mines-collections-add-edit.component.html:157]: i to em tag

**Review by danieltruong — CHANGES_REQUESTED** [review]

File: angular/projects/admin-nrpti/src/app/mines/mines-collection-detail/mines-collection-detail-record-row/mines-collection-detail-record-row.component.html (hunk 2 above)
Comment (KitArmstrong) [reply angular/projects/admin-nrpti/src/app/mines/mines-collection-detail/mines-collection-detail-record-row/mines-collection-detail-record-row.component.html:2]: Discussed. Leaving to only use the link icon.

File: api/src/controllers/search.js (hunk 1 above)
Comment (KitArmstrong) [reply api/src/controllers/search.js:816]: Due to the $unwind above I think we will always need to the group.

File: angular/projects/admin-nrpti/src/app/mines/mines-collection-detail/mines-collection-detail-record-row/mines-collection-detail-record-row.component.html (hunk 2 above)
Comment (KitArmstrong) [reply angular/projects/admin-nrpti/src/app/mines/mines-collection-detail/mines-collection-detail-record-row/mines-collection-detail-record-row.component.html:2]: Updated

File: angular/projects/admin-nrpti/src/app/mines/mines-collections-add-edit/mines-collections-add-edit.component.html (hunk 3 above)
Comment (KitArmstrong) [reply angular/projects/admin-nrpti/src/app/mines/mines-collections-add-edit/mines-collections-add-edit.component.html:157]: Updated

**Review by mtCarto — APPROVED** [review]

**Comment by KitArmstrong** [comment]
> image
> When staging a new records with a link, the list does not show the link icon for the record.

Fixed to handle staged documents.

**Review by danieltruong — APPROVED** [review]

**Pull request merged**